fuel-merkle/src/storage.rs

```diff
@@ -24,7 +23,7 @@ pub trait StorageMutateInfallible<Type: Mappable> {
 
 impl<S, Type> StorageInspectInfallible<Type> for S
 where
-    S: StorageInspect<Type, Error = Infallible>,
+    S: StorageInspect<Type>,
     Type: Mappable,
```

Comment:
The idea of this blanket implementation is that we can unwrap the result of the call because we are sure it is impossible to have an error. Is that true now?

Reply:
I've changed the approach, see https://github.com/FuelLabs/fuel-vm/pull/582#issuecomment-1728570111